src/main/groovy/org/transmartproject/batch/batchartifacts/AbstractMultipleVariablesPerSampleFieldSetMapper.groovy

```diff
@@ -0,0 +1,44 @@
+package org.transmartproject.batch.batchartifacts
+
+import org.springframework.batch.item.file.transform.FieldSet
+import org.springframework.validation.BindException
+
+import java.util.regex.Matcher
+
+/**
+ * Maps the field set to the multiple fields of the multiple items.
+ */
+abstract class AbstractMultipleVariablesPerSampleFieldSetMapper<T> implements MultipleItemsFieldSetMapper<T> {
+
+    private static final COLUMN_NAME_TUPLE_PATTERN = /^(.*)\.([^.]+)$/
+
+    abstract T newInstance(String annotation, String sampleCode)
+
+    abstract Map<String, Closure> getFieldSetters()
```

Comment:
This would need more documentation, for instance what the closures get and should do/return.

But IMO, this design is not very good because it limits the type of T to mutable types. It would be better to just pass all the data to the factory, in a map.